zarr/storage.py

```diff
@@ -1877,6 +1877,141 @@ def __delitem__(self, key):
+        >>> z = zarr.zeros((10, 10), chunks=(5, 5), store=store, overwrite=True)
+        >>> z[...] = 42
+        >>> store.close()  # don't forget to call this when you're done
+
+    Store a group::
+
+        >>> store = zarr.SQLiteStore('data/group.sqldb')
+        >>> root = zarr.group(store=store, overwrite=True)
+        >>> foo = root.create_group('foo')
+        >>> bar = foo.zeros('bar', shape=(10, 10), chunks=(5, 5))
+        >>> bar[...] = 42
+        >>> store.close()  # don't forget to call this when you're done
+    """
+
+    def __init__(self, path, **kwargs):
+        import sqlite3
+
+        kwargs.setdefault('timeout', 5.0)
+        kwargs.setdefault('detect_types', 0)
+        kwargs.setdefault('isolation_level', None)  # autocommit
+        kwargs.setdefault('check_same_thread', False)  # disallow writing from other threads
+        kwargs.setdefault('cached_statements', 100)
+
+        # normalize path
+        if path != ":memory:":
+            path = os.path.abspath(path)
+
+        # store properties
+        self.path = path
+        self.kwargs = kwargs
+
+        # open database
+        self.db = sqlite3.connect(self.path, **self.kwargs)
+
+        # handle keys as `str`s
+        self.db.text_factory = str
+
+        # get a cursor to read/write to the database
+        self.cursor = self.db.cursor()
+
+        # initialize database with our table if missing
+        self.cursor.execute(
+            'CREATE TABLE IF NOT EXISTS kv(k TEXT PRIMARY KEY, v BLOB)'
+        )
+
+    def __getstate__(self):
+        try:
+            self.flush()
+        except Exception:
+            # flush may fail if db has already been closed
+            pass
+        return self.path, self.kwargs
+
+    def __setstate__(self, state):
+        path, kwargs = state
+        self.__init__(path=path, **kwargs)
+
+    def flush(self):
+        self.db.commit()
+
+    def close(self):
+        """Closes the underlying database."""
+
+        # write out everything
+        self.flush()
+
+        # close cursor and db objects
+        self.cursor.close()
+        self.db.close()
+
+    def __getitem__(self, key):
+        for v, in self.cursor.execute('SELECT v FROM kv WHERE k = ?', (key,)):
+            return v
+        else:
+            raise KeyError(key)
+
+    def __setitem__(self, key, value):
+        # Python 2 cannot store `memoryview`s, but it can store `buffer`s.
+        # However Python 2 won't return `bytes` then. So we coerce to `bytes`,
+        # which are handled correctly. Python 3 doesn't have these issues.
+        if PY2:  # pragma: py3 no cover
+            value = ensure_bytes(value)
+        else:  # pragma: py2 no cover
+            value = ensure_contiguous_ndarray(value)
+
+        self.cursor.execute('REPLACE INTO kv VALUES (?, ?)', (key, value))
+
+    def __delitem__(self, key):
+        if key in self:
+            self.cursor.execute('DELETE FROM kv WHERE k = ?', (key,))
+            return
```

Comment:
Is the return statement needed?

Reply:
Good catch. Nope it was left over from some older code. Have dropped it.